### ISSUE TITLE
Reduce package size (again)

### ISSUE THREAD

**Issue by vhiairrassary** [cloud-carbon-footprint/cloud-carbon-footprint#1085]
Total package size has been reduced in https://github.com/cloud-carbon-footprint/cloud-carbon-footprint/pull/276 ([issue](https://github.com/cloud-carbon-footprint/cloud-carbon-footprint/issues/275)). But it seems `googleapis` dependency has been [added again](https://github.com/cloud-carbon-footprint/cloud-carbon-footprint/commit/6b95c629e9f2b52fb1edbb8f52d09ff025fec53f) and it is bigger than before. What do you think about fixing it again?

Output of `@cloud-carbon-footprint/app: 1.1.0`

```bash
ls node_modules/ | xargs -I {} du -h -d0 ./node_modules/{} | grep 'M\s' | sort -k1 -n
1.4M	./node_modules/@sinonjs
1.4M	./node_modules/winston
1.8M	./node_modules/node-forge
2.6M	./node_modules/underscore
2.9M	./node_modules/date-utils
3.0M	./node_modules/bson
3.1M	./node_modules/ramda
3.3M	./node_modules/mongodb
3.4M	./node_modules/protobufjs
4.6M	./node_modules/@cloud-carbon-footprint
4.9M	./node_modules/lodash
5.0M	./node_modules/sinon
5.1M	./node_modules/moment
5.6M	./node_modules/@grpc
5.9M	./node_modules/@types
9.3M	./node_modules/csvtojson
 11M	./node_modules/google-gax
 20M	./node_modules/@google-cloud
 26M	./node_modules/@aws-sdk
 32M	./node_modules/@azure
100M	./node_modules/googleapis
```

**Comment by 4upz** [comment]
Hi @vhiairrassary thanks for looking into this and bring up this issue! It looks like the module was added back to support our implementation of Google Cloud Recommenders. Since the initial commit that you linked to, it seems we've reduced its usage to only be used for type assignments -- specifically for schemas surrounding different machines. These occur within the `packages/gcp/src/lib/ServiceWrapper.ts` and `packages/gcp/src/lib/Recommendations.ts`.

I think it's good to remove if we can replace those types with custom references or with something exported from the `@google-cloud` package like last time. Is this something you'd be interested in taking a stab at again? Otherwise, we're happy to add it to our backlog. Sorry for reversing your work!

**Comment by 4upz** [comment]
After looking into this more, it seems the original package size reduction occurred before we began splitting cloud providers into their own packages. Unfortunately, the GCP package was extracted with more dependencies on the googleapis package than originally thought. Changing it would require a larger rewrite beyond the scope of just the ServiceWrapper and Recommendations files. 

It's definitely possible, but just a larger refactor with more riskier changes. Keeping it in our todo list, but adjusting the size accordingly.

**Comment by vhiairrassary** [comment]
Thank you @4upz for taking time to have a look. It was our conclusion as well, it is not as easy as the first time. In the meantime, as roughly 150MB are coming from typings we are removing them as part of `postinstall` life cycle operation.

⚠️ **We don't recommend anyone to do it** ⚠️ on our side, as our targets have limited storage capacity and because we are only deploying the application without consuming the packages with typescript then it is a viable workaround.

```
> du -h -d0 ./node_modules
359M	./node_modules

> find ./node_modules -name '*.d.ts' -exec du -ch {} + | grep total$
153M	total

> find ./node_modules -name '*.d.ts' -delete

> du -h -d0 ./node_modules
207M	./node_modules
```

**Comment by 4upz** [comment]
Awesome, thanks for the update @vhiairrassary. Glad to see you have a workaround! That's definitely a large size gap. We'll make it a priority to continue to dive into resolving this and will keep posting our progress here.

**Comment by 4upz** [comment]
Addressed this as part of #1126 and awaiting review

**Comment by vhiairrassary** [comment]
Thank you 😍
